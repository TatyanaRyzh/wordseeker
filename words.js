export default [
    "javascript",
    "alert",
    "for",
    "confirm",
    "number",
    "boolean",
    "string",
    "object",
    "array",
    "key",
    "else",
    "length",
    "split",
    "substring",
    "slice",
    "replace",
    "trim",
    "join",
    "reverse",
    "splice",
    "shift",
    "pop",
    "inshift",
    "push",
    "sort",
    "filter",
    "every",
    "some",
    "map",
    "reduce",
    "react",
    "mongodb",
    "trello",
    "github",
    "scrum",
    "skype",
    "mail",
    "sourcetree",
    "windows",
    "framework",
    "date",
    "html",
    "true",
    "false",
    "null",
    "infinity",
    "undefined",
    "prompt",
    "document",
    "event",
    "game",
    "type",
    "developer",
    "mouse",
    "row",
    "cell",
    "head",
    "body",
    "title",
    "switch",
    "while",
    "break",
    "continue",
    "function",
    "style",
    "argument",
    "delete",
    "new",
    "prototype",
    "create",
    "constructor",
    "call",
    "color",
    "font",
    "size",
    "weight",
    "height",
    "variable",
    "variant",
    "family",
    "letter",
    "word",
    "hello",
    "world",
    "text",
    "message",
    "resolve",
    "transform",
    "align",
    "decoration",
    "line",
    "space",
    "cursor",
    "computer",
    "width",
    "margin",
    "padding",
    "display",
    "visibility",
    "capslock",
    "insert",
    "end",
    "home",
    "border",
    "empty",
    "resize",
    "script",
    "code",
    "enter",
    "interface",
    "pascal",
    "basic",
    "cobol",
    "fortran",
    "delphy",
    "java",
    "prolog",
    "Unicode",
    "algorithm",
    "operation",
    "operator",
    "method",
    "parameter",
    "binary",
    "system",
    "error",
    "internet",
    "web",
    "pade",
    "http",
    "block",
    "begin",
    "long",
    "byte",
    "class",
    "compile",
    "compiler",
    "component",
    "control",
    "alternative",
    "tabulation",
    "backspace",
    "pageup",
    "pagedown",
    "escape",
    "pause",
    "lock",
    "printscreen",
    "default",
    "dynamic",
    "regular",
    "expression",
    "formal",
    "global",
    "local",
    "time",
    "goto",
    "identifier",
    "input",
    "output",
    "information",
    "browser",
    "interpreter",
    "iteration",
    "link",
    "loop",
    "logical",
    "integer",
    "real",
    "float",
    "mathematical",
    "arithmetical",
    "number",
    "out",
    "program",
    "protocol",
    "polymorphism",
    "parse",
    "postfix",
    "prefix",
    "position",
    "record",
    "write",
    "read",
    "file",
    "selection",
    "set",
    "stack",
    "option",
    "language",
    "static",
    "structure",
    "semantic",
    "tree",
    "token",
    "lexem",
    "symbol",
    "virtual",
    "visual",
    "destructor",
    "syntax",
    "library",
    "design",
    "animation",
    "float",
    "point",
    "markup",
    "run",
    "recursion",
    "microsoft",
    "net",
    "alpha",
    "beta",
    "version",
    "product",
    "online",
    "offline",
    "commit",
    "release",
    "assembler",
    "attribute",
    "background",
    "image",
    "base",
    "test",
    "download",
    "loader",
    "boot",
    "load",
    "bug",
    "build",
    "install",
    "close",
    "service",
    "origin",
    "pull",
    "branch",
    "index",
    "staging",
    "help",
    "copy",
    "work",
    "value",
    "remote",
    "master",
    "programmer",
    "project",
    "manager",
    "backend",
    "frontend",
    "redux",
    "client",
    "console",
    "clear",
    "open",
    "concatenate",
    "content",
    "modeling",
    "model",
    "debug",
    "debugger",
    "declare",
    "deploy",
    "exception",
    "execute",
    "fatal",
    "label",
    "flag",
    "save",
    "cut",
    "paste",
    "element",
    "hack",
    "hacker",
    "initialize",
    "emulator",
    "high",
    "low",
    "level",
    "fix",
    "search",
    "instruction",
    "integrate",
    "invoke",
    "localization",
    "machine",
    "mask",
    "macro",
    "server",
    "memory",
    "media",
    "microcode",
    "module",
    "name",
    "namespace",
    "operand",
    "optimize",
    "patch",
    "property",
    "python",
    "development",
    "repository",
    "runtime",
    "simulation",
    "socket",
    "source",
    "statement",
    "step",
    "stop",
    "tag",
    "trigger",
    "round",
    "truncate",
    "user",
    "render",
    "root",
    "list",
    "compare",
    "right",
    "left",
    "timezone",
    "year",
    "month",
    "day",
    "week",
    "hour",
    "minute",
    "second",
    "millisecond",
    "random",
    "ceil",
    "floor",
    "decode",
    "selector",
    "parent",
    "child",
    "focus",
    "blur",
    "target",
    "top",
    "find",
    "update",
    "account",
    "count",
    "next",
    "skip",
    "limit",
    "timer",
    "seek",
    "group",
    "match",
    "auto",
    "save",
    "avatar",
    "upgrade",
    "attach",
    "bios",
    "import",
    "export",
    "bold",
    "italic",
    "notebook",
    "device",
    "driver",
    "torrent",
    "seed",
    "peer",
    "cash",
    "cookie",
    "login",
    "password",
    "provider",
    "network",
    "motherboard",
    "flash",
    "edit",
    "folder",
    "editor",
    "exit",
    "command"
    ]
