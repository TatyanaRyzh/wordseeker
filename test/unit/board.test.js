import words from '../../words'
import createBoard from '../../board'

describe('Board', () => {
  const generateRandom = inputSeed => {
    let seed = inputSeed
    return () => {
      seed = Math.pow(seed, 2) % 10000000
      return seed / 10000000
    }
  }

  it('should be generated for 12x12 with any directions', () => {
    const w = 12
    const h = 12
    const directions = [
      [0, 1],
      [-1, 1],
      [-1, 0],
      [-1, -1],
      [0, -1],
      [1, -1],
      [1, 0],
      [1, 1]
    ]
    const random = generateRandom(1234567)

    const board = createBoard(words, { random, w, h, directions })

    expect(board).toMatchSnapshot()
  })

  it('should be generated for 15x15 with any directions', () => {
    const w = 15
    const h = 15
    const directions = [
      [0, 1],
      [-1, 1],
      [-1, 0],
      [-1, -1],
      [0, -1],
      [1, -1],
      [1, 0],
      [1, 1]
    ]
    const random = generateRandom(7654321)

    const board = createBoard(words, { random, w, h, directions })

    expect(board).toMatchSnapshot()
  })

  it('should be generated for 10x10 with no diagonal directions', () => {
    const w = 10
    const h = 10
    const directions = [[0, 1], [-1, 1], [-1, 0], [-1, -1]]
    const random = generateRandom(325793)

    const board = createBoard(words, { random, w, h, directions })

    expect(board).toMatchSnapshot()
  })

  it('should be generated for 20x15 with only horizontal directions', () => {
    const w = 20
    const h = 15
    const directions = [[-1, -1], [1, -1]]
    const random = generateRandom(100001)

    const board = createBoard(words, { random, w, h, directions })

    expect(board).toMatchSnapshot()
  })
})
