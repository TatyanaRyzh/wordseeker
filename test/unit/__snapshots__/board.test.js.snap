// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should be generated for 10x10 with no diagonal directions 1`] = `
Object {
  "square": Array [
    Array [
      "e",
      "m",
      "a",
      "r",
      "g",
      "i",
      "n",
      "v",
      "y",
      "n",
    ],
    Array [
      "d",
      "s",
      "d",
      "p",
      "p",
      "o",
      "d",
      "t",
      "e",
      "l",
    ],
    Array [
      "o",
      "y",
      "l",
      "o",
      "m",
      "a",
      "r",
      "k",
      "u",
      "p",
    ],
    Array [
      "c",
      "l",
      "i",
      "t",
      "h",
      "e",
      "o",
      "o",
      "t",
      "s",
    ],
    Array [
      "o",
      "a",
      "u",
      "s",
      "p",
      "t",
      "e",
      "r",
      "o",
      "d",
    ],
    Array [
      "r",
      "e",
      "b",
      "o",
      "e",
      "l",
      "e",
      "i",
      "o",
      "a",
    ],
    Array [
      "c",
      "r",
      "r",
      "e",
      "i",
      "d",
      "b",
      "m",
      "r",
      "e",
    ],
    Array [
      "i",
      "p",
      "v",
      "f",
      "r",
      "w",
      "e",
      "e",
      "k",
      "h",
    ],
    Array [
      "m",
      "a",
      "c",
      "o",
      "e",
      "a",
      "r",
      "r",
      "a",
      "y",
    ],
    Array [
      "s",
      "t",
      "b",
      "v",
      "e",
      "r",
      "s",
      "i",
      "o",
      "n",
    ],
  ],
  "words": Array [
    "method",
    "array",
    "head",
    "file",
    "property",
    "markup",
    "bios",
    "token",
    "week",
    "version",
    "build",
    "margin",
    "save",
    "microcode",
    "stop",
    "real",
    "border",
    "root",
  ],
}
`;

exports[`Board should be generated for 12x12 with any directions 1`] = `
Object {
  "square": Array [
    Array [
      "y",
      "b",
      "a",
      "c",
      "o",
      "u",
      "n",
      "t",
      "t",
      "l",
      "t",
      "y",
    ],
    Array [
      "a",
      "e",
      "a",
      "l",
      "e",
      "r",
      "t",
      "t",
      "c",
      "l",
      "n",
      "e",
    ],
    Array [
      "d",
      "t",
      "e",
      "v",
      "k",
      "r",
      "o",
      "w",
      "u",
      "u",
      "a",
      "k",
    ],
    Array [
      "i",
      "a",
      "t",
      "i",
      "m",
      "e",
      "r",
      "t",
      "d",
      "p",
      "o",
      "p",
    ],
    Array [
      "n",
      "l",
      "l",
      "a",
      "c",
      "o",
      "l",
      "u",
      "o",
      "e",
      "o",
      "e",
    ],
    Array [
      "f",
      "i",
      "l",
      "e",
      "s",
      "l",
      "e",
      "o",
      "r",
      "c",
      "t",
      "e",
    ],
    Array [
      "i",
      "p",
      "a",
      "d",
      "d",
      "i",
      "n",
      "g",
      "p",
      "a",
      "u",
      "s",
    ],
    Array [
      "n",
      "e",
      "t",
      "h",
      "g",
      "i",
      "e",
      "w",
      "d",
      "e",
      "w",
      "t",
    ],
    Array [
      "i",
      "o",
      "e",
      "f",
      "a",
      "h",
      "s",
      "p",
      "l",
      "o",
      "o",
      "p",
    ],
    Array [
      "t",
      "l",
      "o",
      "a",
      "d",
      "h",
      "u",
      "m",
      "g",
      "n",
      "o",
      "l",
    ],
    Array [
      "y",
      "n",
      "e",
      "p",
      "a",
      "d",
      "e",
      "l",
      "i",
      "a",
      "m",
      "o",
    ],
    Array [
      "t",
      "a",
      "n",
      "w",
      "o",
      "d",
      "e",
      "g",
      "a",
      "p",
      "a",
      "h",
    ],
  ],
  "words": Array [
    "padding",
    "else",
    "pagedown",
    "product",
    "update",
    "beta",
    "font",
    "count",
    "net",
    "work",
    "infinity",
    "long",
    "pade",
    "mail",
    "loop",
    "local",
    "timer",
    "cut",
    "weight",
    "pop",
    "key",
    "out",
    "alert",
    "call",
    "pull",
    "file",
    "day",
    "load",
  ],
}
`;

exports[`Board should be generated for 15x15 with any directions 1`] = `
Object {
  "square": Array [
    Array [
      "w",
      "i",
      "t",
      "u",
      "o",
      "t",
      "a",
      "b",
      "u",
      "l",
      "a",
      "t",
      "i",
      "o",
      "n",
    ],
    Array [
      "h",
      "n",
      "u",
      "e",
      "y",
      "o",
      "l",
      "p",
      "e",
      "d",
      "r",
      "e",
      "a",
      "l",
      "n",
    ],
    Array [
      "s",
      "t",
      "n",
      "e",
      "s",
      "a",
      "b",
      "t",
      "m",
      "o",
      "u",
      "s",
      "e",
      "r",
      "i",
    ],
    Array [
      "t",
      "e",
      "l",
      "i",
      "t",
      "r",
      "i",
      "g",
      "g",
      "e",
      "r",
      "n",
      "o",
      "o",
      "g",
    ],
    Array [
      "e",
      "r",
      "a",
      "p",
      "g",
      "l",
      "d",
      "e",
      "m",
      "a",
      "g",
      "s",
      "h",
      "r",
      "r",
    ],
    Array [
      "p",
      "n",
      "t",
      "u",
      "f",
      "i",
      "s",
      "n",
      "t",
      "t",
      "r",
      "u",
      "e",
      "r",
      "a",
    ],
    Array [
      "y",
      "e",
      "a",
      "o",
      "a",
      "i",
      "r",
      "t",
      "a",
      "u",
      "w",
      "o",
      "r",
      "e",
      "m",
    ],
    Array [
      "d",
      "t",
      "f",
      "r",
      "l",
      "w",
      "e",
      "o",
      "c",
      "y",
      "n",
      "s",
      "e",
      "e",
      "k",
    ],
    Array [
      "l",
      "r",
      "p",
      "g",
      "s",
      "l",
      "r",
      "e",
      "b",
      "h",
      "d",
      "c",
      "i",
      "e",
      "s",
    ],
    Array [
      "t",
      "o",
      "i",
      "m",
      "e",
      "t",
      "o",
      "i",
      "l",
      "u",
      "c",
      "o",
      "e",
      "t",
      "y",
    ],
    Array [
      "t",
      "m",
      "w",
      "v",
      "e",
      "e",
      "c",
      "c",
      "t",
      "i",
      "g",
      "l",
      "b",
      "y",
      "n",
    ],
    Array [
      "h",
      "d",
      "d",
      "n",
      "e",
      "i",
      "n",
      "l",
      "a",
      "e",
      "f",
      "o",
      "a",
      "b",
      "t",
    ],
    Array [
      "g",
      "e",
      "x",
      "i",
      "f",
      "r",
      "i",
      "i",
      "a",
      "l",
      "d",
      "r",
      "i",
      "h",
      "a",
    ],
    Array [
      "i",
      "e",
      "l",
      "t",
      "n",
      "i",
      "o",
      "p",
      "l",
      "s",
      "w",
      "o",
      "r",
      "k",
      "x",
    ],
    Array [
      "r",
      "s",
      "t",
      "p",
      "h",
      "e",
      "i",
      "g",
      "h",
      "t",
      "s",
      "a",
      "c",
      "d",
      "u",
    ],
  ],
  "words": Array [
    "code",
    "group",
    "tabulation",
    "help",
    "trigger",
    "origin",
    "cursor",
    "color",
    "local",
    "driver",
    "seed",
    "syntax",
    "class",
    "error",
    "false",
    "line",
    "margin",
    "mouse",
    "internet",
    "row",
    "seek",
    "point",
    "height",
    "body",
    "deploy",
    "step",
    "base",
    "net",
    "byte",
    "file",
    "fatal",
    "fix",
    "low",
    "real",
    "game",
    "work",
    "write",
    "right",
    "true",
    "empty",
    "out",
    "bug",
  ],
}
`;

exports[`Board should be generated for 20x15 with only horizontal directions 1`] = `
Object {
  "square": Array [
    Array [
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
      "d",
      "n",
      "f",
      "l",
    ],
    Array [
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
      "d",
      "n",
      "f",
    ],
    Array [
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
      "d",
      "n",
    ],
    Array [
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "e",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
      "d",
    ],
    Array [
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "b",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
    Array [
      "d",
      "n",
      "f",
      "l",
      "s",
      "e",
      "e",
      "t",
      "o",
      "d",
      "a",
      "s",
      "t",
      "a",
      "n",
      "b",
      "y",
      "m",
      "d",
      "o",
    ],
  ],
  "words": Array [
    "beta",
  ],
}
`;
